### PR TITLE
BSTOR-73: Fix wrong brand index being displayed

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

File: client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java [hunk 1]
```diff
@@ -129,11 +130,11 @@ public void execute() {
 
     public void updateFromCurrentBrand() {
         Brand newBrand = currentBrand.get();
-        if (newBrand == brand || activeAnimation) {
```
Comment (Elrhino) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java:132]: This caused the problem. It prevented the animations to complete their cycle.

File: client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java [hunk 2]
```diff
@@ -213,13 +213,21 @@ public void f() {
 
     private void handleClick(final GQuery w) {
         Function function = createAnimation(w);
-        if (activeAnimation) {
-            calls.add(function);
-        } else {
-            function.f();
+        String indexOfSelected = w.css("order");
+
+        if (!isActiveBrand(indexOfSelected)) {
```
Comment (Elrhino) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java:218]: This is to remove the animation when you click on a currently selected brand.

File: client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java [hunk 3]
```diff
@@ -173,10 +174,9 @@ public void f() {
                         activeItem.removeClass(sliderResources.style().activeProduct());
                         w.addClass(sliderResources.style().activeProduct());
 
-                        $(elements).attr("style", "transform: scale(1);")
-                                .one(Event.ONCLICK, null, createProductClickHandler());
```
Comment (Elrhino) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/slider/Slider.java:177]: This caused the animation to be called recursively.

**Comment by olafleur** [comment]
Good job! LGTM.

**Pull request merged**